File: src/main/java/orderbook/orderbook/parameters/values/ItemSpecial.java
package orderbook.orderbook.parameters.values;


public enum ItemSpecial {
    @SuppressWarnings("SpellCheckingInspection") MAINHAND,
    OFFHAND,
    NOTHING;


    public String nameLower() {
        return name().toLowerCase();
    }

    public static ItemSpecial matchParameter(String name) {
        try {
            return valueOf(name.toUpperCase());
        }catch(Exception e) {
            return null;
        }

    }
}

File: src/main/java/orderbook/orderbook/parameters/values/ItemSpecial.kt
package orderbook.orderbook.parameters.values

import java.util.Locale

fun itemSpecialFromArgument(arg: String): ItemSpecial? {
    return try {
        ItemSpecial.valueOf(arg.uppercase(Locale.getDefault()))
    } catch (e: Exception) {
        null
    }
}

enum class ItemSpecial {
    MAINHAND, OFFHAND, NOTHING;

    fun nameLower(): String {
        return name.lowercase(Locale.getDefault())
    }
}